resolwe/flow/serializers/relation.py

```diff
@@ -59,22 +57,36 @@ def get_fields(self):
 
         return fields
 
+    def _create_partitions(self, instance, entities):
+        for entity in entities:
+            RelationPartition.objects.create(
+                relation=instance,
+                entity=entity['entity'],
+                label=entity.get('label', None),
+                position=entity.get('position', None),
+            )
+
     def create(self, validated_data):
         """Create ``Relation`` object and add ``Entities``."""
-        # `entities` field is renamed to `positioninrelation_set` based on source of nested serializer
+        # `entities` field is renamed to `relationpartition_set` based on source of nested serializer
+        entities = validated_data.pop('relationpartition_set', {})
 
-        entities = validated_data.pop('positioninrelation_set', {})
+        with transaction.atomic():
+            instance = Relation.objects.create(**validated_data)
+            self._create_partitions(instance, entities)
 
-        # Prevent "Failed to populate slug Relation.slug from name" output
-        validated_data['name'] = 'Relation'
+        return instance
+
+    def update(self, instance, validated_data):
+        # `entities` field is renamed to `relationpartition_set` based on source of nested serializer
+        entities = validated_data.pop('relationpartition_set', {})
+
+        instance = super().update(instance, validated_data)
```

Comment:
Why not have this in `transaction.atomic()`?

Reply:
Good point :)